server/models/User.coffee

```diff
@@ -445,14 +464,18 @@ UserSchema.pre('save', (next) ->
   next()
 )
 
-UserSchema.post 'save', (doc) ->
-  doc.newsSubsChanged = not _.isEqual(_.pick(doc.get('emails'), mail.NEWS_GROUPS), _.pick(doc.startingEmails, mail.NEWS_GROUPS))
-  UserSchema.statics.updateServiceSettings(doc)
-
+UserSchema.post 'save', co.wrap ->
+  try
+    yield @updateServiceSettings()
+  catch e
+    console.error 'User Post Save Error:', e.stack
 
-UserSchema.post 'init', (doc) ->
-  doc.wasTeacher = doc.isTeacher()
-  doc.startingEmails = _.cloneDeep(doc.get('emails'))
+  
+UserSchema.post 'init', ->
+  @set('anonymous', false) if @get('email')
```

Comment:
I am reluctant to change this behavior, as it's far reaching and orthogonal to the PR. I'd rather (eventually) get rid of this line and rely on signup endpoints to properly make sure `anonymous` is set correctly. Will make a TODO there, once we stop using PUT user to sign users up.